seurat-find-transfer-anchor.R

```diff
@@ -0,0 +1,220 @@
+    type = 'character',
+    help ="Method for nearest neighbor finding. Options include: rann, annoy"
+  ),
+  make_option(
+    c("--eps"),
+    action = "store",
+    default = 0,
+    type = 'integer',
+    help ="Error bound on the neighbor finding algorithm (from RANN)"
+  ),
+  make_option(
+    c("--approx-pca"),
+    action = "store_true",
+    default = TRUE,
+    help ="Use truncated singular value decomposition to approximate PCA"
+  ),
+  make_option(
+    c("--verbose"),
+    action = "store_true",
+    default = TRUE,
+    help ="Print progress bars and output"
+  )
+  
+)
+
+#minimum arguments to work
+opt <- wsc_parse_args(option_list, mandatory = c('query_file', 'reference_file','output_file'))
+
+# Check parameter values
+
+if ( ! file.exists(opt$reference_file)){
+  stop((paste('File', opt$reference_file, 'does not exist')))
+}
+
+
+#load seurat and packages needed to read input
+suppressPackageStartupMessages(require(Seurat))
+if(opt$query_format == "loom" | opt$output_format == "loom") {
+  suppressPackageStartupMessages(require(loomR))
+} else if(opt$query_format == "singlecellexperiment" | opt$output_format == "singlecellexperiment") {
+  suppressPackageStartupMessages(require(scater))
+}
+
+seurat_query <- read_seurat3_object(input_path = opt$reference_file, format = opt$reference_format)
+seurat_reference <- read_seurat3_object(input_path = opt$query_file, format = opt$query_format)
+
+#make the fonction work
+anchor_object <- FindTransferAnchors(seurat_reference,
+                                    seurat_query,
+                                    normalization.method = opt$normalization.method,
+                                    reference.assay = opt$reference.assay,
+                                    query.assay = opt$query.assay,
+                                    reduction = opt$reduction,
+                                    project.query = opt$project.query,
+                                    features = opt$features,
+                                    npcs = opt$npcs,
+                                    l2.norm = opt$l2.norm,
+                                    dims = opt$dims,
+                                    k.anchor = opt$k.anchor,
+                                    k.filter = opt$k.filter,
+                                    k.score = opt$k.score,
+                                    max.features = opt$max.features,
+                                    nn.method = opt$nn.method,
+                                    eps = opt$eps,
+                                    approx.pca = opt$approx.pca,
+                                    verbose = opt$verbose)
+#get the addr for output
+#position <- string_locate_all("/",$output_file)
+#position <- position[position(lenght(position))]
+#display(position)
+#get the anchor matrix and save it as csv
+anchor_matrix <- anchor_object.anchors
+csv_anchor <- writematrix(csv_anchor,paste(output_file(:position), "anchor_matrix.csv"))
```

Comment:
Is CSV the best way to do this? It it easy to reconstruct the AnchorSet object from the CSV in processes that consume this output? It may be easier to output a serialised R object using saveRDS instead.

Reply:
maybe I should use directly TransferData and AddMetaData to just add it to the query like in the exemple

![image](https://user-images.githubusercontent.com/43669368/76779665-a09a0500-67a3-11ea-9b48-64efc270223c.png)

Reply:
So, from that example I can see that TransferData() takes this AnchorSet. For now you should just make sure that your script outputs an AnchorSet directly, rather than a CSV. You'll eventually write a wrapper for TransferData(), and you don't want to have to reconstruct your AnchorSet from text-format outputs.

Reply:
saveRDS() is the function we use to save R objects to disk.